### PR TITLE
update streamsets tutorial with latest jdbc and crate versions

### PR DESCRIPTION
**Pull request by kovrus**
image

image

### CODE REVIEW

File: docs/tools/streamsets.txt [hunk 1]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
-   .. image:: 3-streamsets-dst-driver.png
-      :alt: Load the CrateDB JDBC driver
+   .. image:: 3-streamsets-dst-driver-conf.png
+      :alt: Configure the CrateDB JDBC Producer
 
-   Then, we configure the JDBC connection (1) and define the destination table
-   (2). After the above-mentioned configurations steps, we still can see
-   warnings in the *Credential tab*. It is solved by unchecking the user
-   authorization box because CrateDB does not support an authorization. (3)
+   Finally, we set the default CrateDB in the credentials tab.
```
Comment (MarkusH) [inline docs/tools/streamsets.txt]:
```suggestion
   Finally, we set the default credentials to CrateDB in the "credentials" tab.
```

File: docs/tools/streamsets.txt [hunk 2]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
-   <https://streamsets.com/documentation/datacollector/latest/help/#Pipeline_Design/What_isa_Pipeline.html>`_
-   and configure the CrateDB JDBC Driver loading as it was done for the JDBC
-   destination configuration. In the JDBC tab of the CrateDB JDBC origin we set
-   the connection string (1), the SQL query for fetching the data from the
-   database (2), the offset value and column (3). For more detail information
-   on how to set the offset value and column see the StreamSets `documentation
-   <https://streamsets.com/documentation/datacollector/1.1.3/help/#Origins/JDBCConsumer.html#concept_nxz_2kz_bs>`_.
-
-   .. image:: 3-streamsets-source-jdbc.png
+the data stream pipeline. As sample data, we use `AMPLab rankings dataset`_.
```
Comment (MarkusH) [inline docs/tools/streamsets.txt]:
```suggestion
the data stream pipeline. As sample data, we use the `AMPLab rankings dataset`_.
```

File: docs/tools/streamsets.txt [hunk 3]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+The data can be imported from `AWS S3`_ to the CrateDB database using prepared
+data import queries hosted in the `CrateDB demo repository`_. Create the
+`rankings table`_ and import the `rankings data`_. In the demo, we use the
```
Comment (MarkusH) [inline docs/tools/streamsets.txt]:
```suggestion
`rankings table`_ and import the `rankings data`_. In the demo, we use a
```

File: docs/tools/streamsets.txt [hunk 4]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+dataset that contains 18 millions of records. Having the CrateDB cluster set up
+and the rankings sample data is imported we can start building the data stream
```
Comment (MarkusH) [inline docs/tools/streamsets.txt]:
```suggestion
and the rankings sample data is imported, we can start building the data stream
```

File: docs/tools/streamsets.txt [hunk 5]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+pipeline for streaming the data from the CrateDB database to CSV files.
+
+1. Create a `new data collector pipeline`_ and configure the CrateDB JDBC Driver
+   loading as it was done for the JDBC destination configuration. In the JDBC
+   tab of the CrateDB JDBC origin we set the connection string, the SQL query
+   for fetching the data from the database, the initial offset and offset
```
Comment (MarkusH) [inline docs/tools/streamsets.txt]:
```suggestion
   for fetching the data from the database, the initial offset, and offset
```

File: docs/tools/streamsets.txt [hunk 6]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
-
-2. For the sake of simplicity, we use only 4 fields from the CSV file. The
-   directory origin does not perform any preliminary preprocessing of
-   records. All source fields are represented as strings in the CSV.
-   However, some of the selected fields should be processed and
-   converted into float values. Therefore, we add an additional
-   processing `Field Converter
-   <https://streamsets.com/documentation/datacollector/latest/help/#Processors/FieldConverter.html#concept_is3_zkp_wq>`_ stage into the
-   pipeline.
+The next two sections provide a brief introduction to building data stream
```
Comment (chaudum) [inline docs/tools/streamsets.txt]:
```suggestion
The next two sections provide a brief introduction on how to build data stream
```

File: docs/tools/streamsets.txt [hunk 7]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+pipelines using CrateDB with StreamSets Data Collector. In the first section, we
+are going to demonstrate how to build the pipeline with the `directory origin`_
+stage that supplies the CSV data sample and streams data to JDBC destination
+stage using the CrateDB JDBC driver. In the second section, we demonstrate how
+to use the CrateDB JDBC driver as a StreamSets origin.
+
+We use the following versions in the tutorial:
+
+- CrateDB - `3.2.3`
```
[the remark this answers is not in the document]
Comment (chaudum) [reply docs/tools/streamsets.txt]:
```suggestion
- CrateDB - ``3.2.3``
```

File: docs/tools/streamsets.txt [hunk 8]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+- CrateDB JDBC Driver - `2.5.1`
```
[the remark this answers is not in the document]
Comment (chaudum) [reply docs/tools/streamsets.txt]:
```suggestion
- CrateDB JDBC Driver - ``2.5.1``
```

File: docs/tools/streamsets.txt [hunk 9]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+- StreamSets Data Collector - `3.7.2`
```
[the remark this answers is not in the document]
Comment (chaudum) [reply docs/tools/streamsets.txt]:
```suggestion
- StreamSets Data Collector - ``3.7.2``
```

File: docs/tools/streamsets.txt [hunk 10]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+The next two sections provide a brief introduction to building data stream
+pipelines using CrateDB with the StreamSets Data Collector. In the first
+section, we are going to demonstrate how to build the pipeline with the
+`directory origin`_ stage that supplies the CSV data sample and streams data
+to JDBC destination stage using the CrateDB JDBC driver. In the second section,
+we demonstrate how to use the CrateDB JDBC driver as a StreamSets origin.
+
+We use the following versions in the tutorial:
+
+- CrateDB – `3.2.3`
```
Comment (chaudum) [inline docs/tools/streamsets.txt]:
```suggestion
- CrateDB – ``3.2.3``
```

File: docs/tools/streamsets.txt [hunk 11]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+- CrateDB JDBC Driver – `2.5.1`
```
Comment (chaudum) [inline docs/tools/streamsets.txt]:
```suggestion
- CrateDB JDBC Driver – ``2.5.1``
```

File: docs/tools/streamsets.txt [hunk 12]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+- StreamSets Data Collector – `3.7.2`
```
Comment (chaudum) [inline docs/tools/streamsets.txt]:
```suggestion
- StreamSets Data Collector – ``3.7.2``
```

File: docs/tools/streamsets.txt [hunk 13]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
+
+CrateDB JDBC Producer
+=====================
+
+The CrateDB JDBC Producer stage uses the JDBC connection to write data into the
+database. In this section, we show how to build the StreamSets project for
+injecting CSV records from the local filesystem into the CrateDB database with
```
Comment (chaudum) [inline docs/tools/streamsets.txt]:
```suggestion
ingesting CSV records from the local filesystem into CrateDB with
```

File: docs/tools/streamsets.txt [hunk 14]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
-CrateDB JDBC Origin
-===================
+CrateDB JDBC Query Consumer
+===========================
 
-The JDBC Consumer origin uses the JDBC connection to read data from the CrateDB
-database. The CrateDB origin stage returns data as a map with column names and
-field values.
+The `JDBC Query Consumer`_ uses the JDBC connection to read data from the
+CrateDB database. The CrateDB JDBC Query Consumer stage returns data as a map
```
Comment (chaudum) [inline docs/tools/streamsets.txt]: from CrateDB

**Comment by kovrus** [comment]
@chaudum addressed your comments

File: docs/tools/streamsets.txt [hunk 15]
```diff
@@ -10,168 +10,159 @@ Building Data Stream pipelines with CrateDB and StreamSets Data Collector
...
 
-   .. image:: 6-streamsets-origin-report.png
+   .. image:: 4-streamsets-origin-report.png
       :alt: CrateDB to CSV report
 
--  `Download StreamSets <https://streamsets.com/opensource/>`_
--  `StreamSets Official documentation
-   <https://streamsets.com/documentation/datacollector/latest/help/>`_
+
+.. _`CrateDB`: https://crate.io
```
Comment (mechanomi) [inline docs/tools/streamsets.txt]: please sort this whole list alphabetically

Comment (mechanomi) [reply docs/tools/streamsets.txt]: please also drop the ` from around the link titles. they're not needed in the link list

**Comment by kovrus** [comment]
@mechanomi done

**Pull request merged**